dashboard/app/models/pd/workshop.rb

```diff
@@ -345,6 +345,13 @@ def friendly_date_range
       "#{sessions.first.start.strftime('%B %-d')} - #{sessions.last.start.strftime('%B %-d, %Y')}"
   end
 
+  def date_and_location_name
+    date_string = sessions.any? ? friendly_date_range : 'Dates TBA'
```

Comment:
should `friendly_date_range` itself be updated to deal with no sessions?

Reply:
Later change

Reply:
Note the UI doesn't allow creating a workshop with no sessions, so generally speaking this can't happen in the wild. However the model does and the default factories don't create sessions, so many tests use workshops with no sessions.